modules/log4j-configurator/src/main/java/org/mule/runtime/module/log4j/internal/ApplicationReconfigurableLoggerContextSelector.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.log4j.internal;
+
+import static org.mule.runtime.deployment.model.internal.artifact.CompositeClassLoaderArtifactFinder.findClassLoader;
+
+import static java.lang.ClassLoader.getSystemClassLoader;
+import static java.util.Arrays.asList;
+
+import org.mule.runtime.core.internal.util.CompositeClassLoader;
+import org.mule.runtime.deployment.model.api.policy.PolicyTemplateDescriptor;
+import org.mule.runtime.module.artifact.activation.internal.classloader.MuleSharedDomainClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.classloader.RegionClassLoader;
+
+import java.net.URI;
+import java.util.List;
+
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.selector.ContextSelector;
+
+/**
+ * A {@link ContextSelector} that will use the context for the application once it is deployed. It may be used in environments
+ * that deploys only one app (W-12356154).
+ *
+ * @since 4.7.0
+ */
+public class ApplicationReconfigurableLoggerContextSelector implements ContextSelector {
+
+  private final MuleLoggerContextFactory loggerContextFactory = new MuleLoggerContextFactory();
+  private static final ClassLoader SYSTEM_CLASSLOADER = getSystemClassLoader();
+  private LoggerContext containerLoggerContext;
```

Comment:
`container` in this context might be confusing

Reply:
removed the string "container"